common/tools/dev-tool/src/commands/samples/checkNodeVers.ts

```diff
@@ -0,0 +1,323 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import fs from "fs-extra";
+import path from "path";
+import pr from "child_process";
+import os from "os";
+
+import { createPrinter } from "../../util/printer";
+import { leafCommand, makeCommandInfo } from "../../framework/command";
+import { S_IRWXO } from "constants";
+import { resolveProject } from "../../util/resolveProject";
+
+const log = createPrinter("check-node-vers-samples");
+
+function spawnCMDIgnoreError(
```

Comment:
I don't understand this function. It says it will `IgnoreError` but has logic in it to print error messages. I'm not a huge fan of letting errors pass by in commands that run on our dev boxes, but I could be convinced that the risk is low.

Also, can this be made async by `new`ing up a Promise and using resolve/reject in place of the callback calls?

Reply:
I am not sure what is your concern? This function proceeds with calling the callback even in the presence of errors and is used for cleanup operations so it does not really matter if the cleanup failed.

Reply:
A man loves his callbacks ...
Done.